### DISCUSSION TITLE
Digital products delivery

### DISCUSSION THREAD

**Discussion by tanujrawat95** [saleor/saleor#12252] [Q&A]
Hi everyone!

I am trying to make a platform where customers can order a report based on their input.
That means the final product will be different for every order.
I am trying to understand the flow after the customer pay for the product.

For my use case, I will required to send the report/product or the link to download to the user via mail and storefront. 
As far as i think, I can make another workflow to upload the report to s3 and deliver the report to customers via mail and update the order metadata by graphql query so that it will get reflect in customer's dashboard. 

Is there any better way to handle this?

**Comment by timuric** [comment]
Hey @tanujrawat95 this is a pretty common and flexible flow

**Reply by tanujrawat95** [reply]
Hi @timuric ! thanks for the response. So the solution that i could think of will be the best approach for it right? But I have checked, there is no feature to update metadata for order or orderline through graphql mutations -
[orderLineUpdate](https://docs.saleor.io/docs/3.x/api-reference/mutations/order-line-update)
[orderUpdate](https://docs.saleor.io/docs/3.x/api-reference/mutations/order-update)

**Reply by tanujrawat95** [reply]
I guess "updateMetadata" mutation will work for that. 

Also can you please help me for invoice generation?  
1.  I want them to generate automatically when order get paid successfully
2.  I want to store them in S3 bucket
